layer.go

```diff
@@ -154,6 +154,12 @@ func (l *LayerContributor) normalizeDependencyDeprecationDate(input map[string]i
 				break
 			}
 		}
+	} else if depr_date, ok := input["deprecation_date"]; ok {
+		deprecationDate, err := l.parseDeprecationDate(depr_date)
```

Comment:
Minor nit pick, but maybe put this into a function cause it's being used here and on line 149 above?